### ISSUE TITLE
标签数据

### ISSUE THREAD

**Issue by hongminglin08** [LiQiang0307/CLTDR-GMG#2]
您好，我尝试跑了一下THUMOS14、MultiTHUMOS、EPIC-KITCHENS-100 和 ActivityNet-1.3，实验配置也跟本文的一致，但实验结果基本都远低于这篇论文报告的结果，尤其是THUMOS14、MultiTHUMOS和ActivityNet-1.3，分别只能达到71%、37%和34%的Avearge mAP（原文中应该能够达到74.3%、44.9和37.1%）。是否代码有缺失什么关键部分吗？或者您是否方便提供一下annotation数据？谢谢

**Comment by LiQiang0307** [comment]
请检查使用的特征是否一致，实验中用到了不同backbone （i3d 、videomaev2、intervideo ）提取的特征 。此外，不同数据集在不同loss（focalloss、vflloss）下结果会有所不同。

**Comment by hongminglin08** [comment]
我确实是按照论文说明的backbone进行实验的，例如MultiTHUMOS就是使用的videomaev2提取的特征。但远没办法达到论文的效果。损失的使用我是直接使用代码，没有经过修改，（代码部分应该是使用0.7focalloss加上0.3的vflloss）。我查看原论文，好像没有说明不同数据集应该怎么选取loss，请问应当怎么根据不同数据集进行修改

![Image](https://github.com/user-attachments/assets/b64bf120-9867-4243-9f42-edf868c3e792)

**Comment by LiQiang0307** [comment]
# focal loss sigmoid_focal_loss/varifocal_loss
        cls_loss = sigmoid_focal_loss(
            torch.cat(out_cls_logits, dim=1)[valid_mask],
            gt_target,
            reduction='none'
        )

[meta_archs.py](https://github.com/LiQiang0307/CLTDR-GMG/blob/main/libs/modeling/meta_archs.py) 585行，请根据配置文件后缀，设置cls_loss是使用sigmoid_focal_loss还是varifocal_loss

**Comment by hongminglin08** [comment]
> # focal loss sigmoid_focal_loss/varifocal_loss
> ```
>     cls_loss = sigmoid_focal_loss(
>         torch.cat(out_cls_logits, dim=1)[valid_mask],
>         gt_target,
>         reduction='none'
>     )
> ```
> 
> [meta_archs.py](https://github.com/LiQiang0307/CLTDR-GMG/blob/main/libs/modeling/meta_archs.py) 585行，请根据配置文件后缀，设置cls_loss是使用sigmoid_focal_loss还是varifocal_loss

可以了，感谢

**Comment by lixueli8** [comment]
> > # focal loss sigmoid_focal_loss/varifocal_loss
> > ```
> >     cls_loss = sigmoid_focal_loss(
> >         torch.cat(out_cls_logits, dim=1)[valid_mask],
> >         gt_target,
> >         reduction='none'
> >     )
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [meta_archs.py](https://github.com/LiQiang0307/CLTDR-GMG/blob/main/libs/modeling/meta_archs.py) 585行，请根据配置文件后缀，设置cls_loss是使用sigmoid_focal_loss还是varifocal_loss
> 
> 可以了，感谢

你好，请问你复现出论文里面的结果了嘛，我复现的代码 thumos14-intervideo特征 （用varifocal_loss）最高只达到73.64 mAP；thumos14-I3D特征 （用sigmoid_focal_loss）最高只达到68.70 mAP； thumos14-videoMAE特征 （用varifocal_loss）最高达到71.4 mAP.
